src/data/data.py

```diff
@@ -222,7 +222,7 @@ def setup(self):
         self.val_samples = shuffle(self.val_samples, random_state=42)
         self.test_samples = shuffle(self.test_samples, random_state=42)
 
-        self.train_dataset = DaVinciDataSet(data_dir=self.data_dir,
+        self.train_dataset = self.dataset(data_dir=self.data_dir,
```

Comment:
any specific reason for doing this?

Reply:
This allows us to reuse the base class `setup()` rather than overwriting it for all subclasses (since they use different DataSet classes)

Reply:
niiiice